File: src/main/java/com/fretron/terminalModels/TripSh.java
package com.fretron.terminalModels;

import com.google.gson.GsonBuilder;

public class TripSh {

    private String uuid;
    private String shipmentNumber;
    private String externalId;
    private String fuItemId;
    private ResourceInfo pickedContainers;
    private ResourceInfo pickedCn;
    private ResourceInfo origin;
    private ResourceInfo destination;
    private ResourceInfo tripInfo;
    private String containerizationType;

    public TripSh() {

    }

    public TripSh(String uuid, String shipmentNumber, String externalId, String fuItemId, ResourceInfo pickedContainers,
                  ResourceInfo pickedCn, ResourceInfo origin, ResourceInfo destination, ResourceInfo tripInfo,
                  String containerizationType) {
        this.uuid = uuid;
        this.shipmentNumber = shipmentNumber;
        this.externalId = externalId;
        this.fuItemId = fuItemId;
        this.pickedContainers = pickedContainers;
        this.pickedCn = pickedCn;
        this.origin = origin;
        this.destination = destination;
        this.tripInfo = tripInfo;
        this.containerizationType = containerizationType;
    }

    public String getUuid() {
        return uuid;
    }

    public void setUuid(String uuid) {
        this.uuid = uuid;
    }

    public String getShipmentNumber() {
        return shipmentNumber;
    }

    public void setShipmentNumber(String shipmentNumber) {
        this.shipmentNumber = shipmentNumber;
    }

    public String getExternalId() {
        return externalId;
    }

    public void setExternalId(String externalId) {
        this.externalId = externalId;
    }

    public String getFuItemId() {
        return fuItemId;
    }

    public void setFuItemId(String fuItemId) {
        this.fuItemId = fuItemId;
    }

    public ResourceInfo getPickedContainers() {
        return pickedContainers;
    }

    public void setPickedContainers(ResourceInfo pickedContainers) {
        this.pickedContainers = pickedContainers;
    }

    public ResourceInfo getPickedCn() {
        return pickedCn;
    }

    public void setPickedCn(ResourceInfo pickedCn) {
        this.pickedCn = pickedCn;
    }

    public ResourceInfo getOrigin() {
        return origin;
    }

    public void setOrigin(ResourceInfo origin) {
        this.origin = origin;
    }

    public ResourceInfo getDestination() {
        return destination;
    }

    public void setDestination(ResourceInfo destination) {
        this.destination = destination;
    }

    public ResourceInfo getTripInfo() {
        return tripInfo;
    }

    public void setTripInfo(ResourceInfo tripInfo) {
        this.tripInfo = tripInfo;
    }

    public String getContainerizationType() {
        return containerizationType;
    }

    public void setContainerizationType(String containerizationType) {
        this.containerizationType = containerizationType;
    }

    @Override
    public String toString() {
        return new GsonBuilder().serializeNulls().create().toJson(this);
    }
}

File: src/main/java/com/fretron/terminalModels/TripSh.kt
package com.fretron.terminalModels

import com.fretron.Model.BuisnessPartner
import com.fretron.Model.ResourceAlert
import com.google.gson.GsonBuilder

data class TripSh(
    var uuid: String?,
    var shipmentNumber: String?,
    var shipmentDate : Long? ,
    var externalId: String?,
    var orgId: String?,
    var fuItemId: String?,
    var stages: MutableList<TripShStage>,
    var tripInfo: ResourceInfo?,
    var containerizationType: String?,
    var partners: List<PartnerInfo>,
    var status: String?,
    var isTripDTChanged : Boolean,
    var alerts : List<ResourceAlert>,
    var shipmentCostNumber : String?
) {

    constructor() : this(
        null, null , null, null, null, null, mutableListOf(), null,
        null, emptyList(), null , false , mutableListOf(),null
    )

    fun toResourceInfo(): ResourceInfo {
        return ResourceInfo().also {
            it.resourceId = uuid
            it.resourceIdentifier = shipmentNumber
            it.resourceType = "SHIPMENT"
            it.resourceExtId = externalId
        }
    }

    fun toShipmentInfo(): ShipmentInfo {
        return ShipmentInfo().also {
            it.resourceId = uuid
            it.resourceIdentifier = shipmentNumber
            it.resourceExtId = externalId
            it.vehicleRegistrationNumber = null
            if (partners.firstOrNull()?.partner != null) {
                val p = partners.first().partner!!
                it.vendor = LiteBusinessPartner().apply {
                    this.uuid = p.getUuid()
                    this.externalId = p.getExternalId()
                    this.placeId = if (p.getPlaces().isNullOrEmpty()) null else p.getPlaces()[0]?.getPlaceId()
                    this.placeName = if (p.getPlaces().isNullOrEmpty()) null else p.getPlaces()[0]?.getName()
                    this.placeExternalId =
                        if (p.getPlaces().isNullOrEmpty()) null else p.getPlaces()[0]?.getExternalId()
                    this.name = p.getName()
                    this.address = if (p.getPlaces().isNullOrEmpty()) null else p.getPlaces()[0]?.getAddress()
                }

            }

        }
    }

    override fun toString(): String {
        return GsonBuilder().serializeNulls().create().toJson(this)
    }
}

data class TripShStage(
    var type: String?,
    var terminalId: String?,
    var terminalName: String?,
    var terminalExtId: String?,
    var equipmentsPickups: MutableList<ResourceInfo>,
    var equipmentsDropped: MutableList<ResourceInfo>,
    var consignmentPickups: MutableList<LiteConsignment>,
    var consignmentDropped: MutableList<LiteConsignment>

) {
    constructor() : this(null, null, null, null, mutableListOf(), mutableListOf(), mutableListOf(), mutableListOf())

    override fun toString(): String {
        return GsonBuilder().serializeNulls().create().toJson(this)
    }
}

data class PartnerInfo(var role: String?, var partner: BuisnessPartner?) {
    constructor() : this(null, null)

    override fun toString(): String {
        return GsonBuilder().serializeNulls().create().toJson(this)
    }
}
